File: ANewFolder/MyCards-master(1)/MyCards-master/app/src/main/java/edu/univdhaka/cse/cse2216/mycards/java/repository/CardRepository.java
package edu.univdhaka.cse.cse2216.mycards.java.repository;

import java.util.List;

import edu.univdhaka.cse.cse2216.mycards.java.domain.Card;

public interface CardRepository {

    List<Card> listCards();

    void addCard(Card card);

    void updateCard(Card card);

    void deleteCard(Card card);

}

File: ANewFolder/MyCards-master (2)/app/src/main/java/edu/univdhaka/cse/cse2216/mycards/kotlin/repository/CardRepository.kt
package edu.univdhaka.cse.cse2216.mycards.kotlin.repository

import edu.univdhaka.cse.cse2216.mycards.kotlin.domain.Card

interface CardRepository {

    fun listCards(): List<Card>

    fun listCardsAsync(onResult: (List<Card>) -> Unit)

    fun addCard(card: Card)

    fun updateCard(card: Card)

    fun deleteCard(card: Card)

}
